### PR TITLE
Homework5: Aleksandr Levdansky

### PR DESCRIPTION
**Pull request by Coromilly**

### CODE REVIEW

File: years.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
+2017/I #
```
Comment (belovmd) [inline years.txt]: what does this year means?

File: runner.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    else:
+        for func_name in args:
+            if hasattr(pytasks, func_name):
+                method_to_call = getattr(pytasks, func_name)
+                method_to_call()
+            else:
+                print(func_name, 'Name doesn\'t exist')
+
+
+# Uncomment to use the second solution
```
Comment (belovmd) [inline runner.py]: do we need these comments?

File: imdb.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+try:
+    with open(file_path) as ratings_list:
+        top250 = open('top20_movies.txt', 'w')
+        rating_histogram = open('rating.txt', 'w')
+        years_histogram = open('years.txt', 'w')
+        rating = []
+        years = []
+        for num, line in enumerate(ratings_list):
+            if num in range(28, 278):
+                top250.write(' '.join(line.split()[3:-1]) + '\n')
```
Comment (belovmd) [inline imdb.py]: too many magic items 3 1 2

File: imdb.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+path = input('Please input full path to "ratings.list" file:')
+if path[-1] == '\\':
+    file_path = os.path.join(path, "ratings.list")
+else:
+    file_path = os.path.join(path, "\\ratings.list")
+try:
+    with open(file_path) as ratings_list:
+        top250 = open('top20_movies.txt', 'w')
+        rating_histogram = open('rating.txt', 'w')
```
Comment (belovmd) [inline imdb.py]: why can't we use context manager?

File: runner.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import pytasks
+
+
+def runner(*args):
+    """Function prints result of args taken function."""
+    if not args:
+        pytasks.main()
```
Comment (belovmd) [inline runner.py]: it shouldn't find main it should run all

File: runner.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import pytasks
```
Comment (belovmd) [inline runner.py:14]: missed task description

**Review by belovmd — COMMENTED** [review]
please address these issues

**Comment by belovmd** [comment]
histograms shouldn't contain 2017/I such years

File: years.txt (hunk 1 above)
Comment (Coromilly) [reply years.txt]: i don't know
it's in top250 list

File: runner.py (hunk 2 above)
Comment (Coromilly) [reply runner.py]: fixed.

File: runner.py (hunk 6 above)
Comment (Coromilly) [reply runner.py:14]: solved.

File: runner.py (hunk 5 above)
Comment (Coromilly) [reply runner.py]: solved.

File: imdb.py (hunk 4 above)
Comment (Coromilly) [reply imdb.py]: fixed.

File: imdb.py (hunk 3 above)
Comment (Coromilly) [reply imdb.py]: fixed.

File: imdb.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                year_start_pos = 1
+                year_end_pos = 5
+                years.append(line.split()[years_column][year_start_pos:
+                                                        year_end_pos])
+except FileNotFoundError:
+    print("File \"ratings.list\" doesn\'t exist")
+
+with open('top20_movies.txt', 'w') as top250:
+    for film in films:
+        top250.write(film + '\n')
```
Comment (belovmd) [inline imdb.py:39]: it would be redundant \\n in the end of file

File: imdb.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+with open('rating.txt', 'w') as rating_histogram:
+    for rate in sorted(set(rating), reverse=True):
+        rating_histogram.write(rate + ' ' + '#' * rating.count(rate) + '\n')
```
Comment (belovmd) [inline imdb.py:42]: same

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**